src/main/java/com/google/vinet/servlets/RequestServlet.java

```diff
@@ -0,0 +1,360 @@
+    if (response == null) {
+      throw new IllegalArgumentException("response must not be null");
+    }
+
+    if (request == null) {
+      throw new IllegalArgumentException("request must not be null");
+    }
+
+    if (!this.userService.isUserLoggedIn()) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be logged in to post a request"
+      );
+      return;
+    }
+
+    final boolean registered = RegistrationServlet.isUserRegistered(this.userService);
+
+    if (!registered) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be registered to post a request"
+      );
+      return;
+    }
+
+    final boolean isIsolate = RegistrationServlet.isUserIsolate(this.userService);
+
+    if (!isIsolate) {
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "user must be registered as an isolate to post a request"
+      );
+      return;
+    }
+
+    String date = request.getParameter("date");
+    String duration = request.getParameter("duration");
+    String startTime = request.getParameter("startTime");
+    String endTime = request.getParameter("endTime");
+    String timezoneOffset = request.getParameter("timezoneOffset");
+    String[] subjects = request.getParameterValues("subject");
+    String[] details = request.getParameterValues("details");
+
+    if (date == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "date must not be null"
+      );
+      return;
+    }
+
+    if (duration == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "duration must not be null"
+      );
+      return;
+    }
+
+    if (startTime == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "startTime must not be null"
+      );
+      return;
+    }
+
+    if (endTime == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "endTime must not be null"
+      );
+      return;
+    }
+
+    if (timezoneOffset == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "timezoneOffset must not be null"
+      );
+      return;
+    }
+
+    if (subjects == null || subjects.length < 1) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "subjects array must not be null or empty"
+      );
+      return;
+    }
+
+    if (details == null || details.length < 1) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "details array must not be null or empty"
+      );
+      return;
+    }
+
+    date = date.trim();
+    duration = duration.trim();
+    startTime = startTime.trim();
+    endTime = endTime.trim();
+    timezoneOffset = timezoneOffset.trim();
+    subjects = trimMembers(subjects);
+    details = trimMembers(details);
+
+    if (date.equals("")) {
```

Comment:
Could this not use String.isEmpty() ?

Reply:
Not sure why I never considered that, I think I'll change it over now